lib/trino-filesystem/src/main/java/io/trino/filesystem/hdfs/HdfsFileSystem.java

```diff
@@ -75,6 +83,30 @@ public void deleteFile(String path)
         });
     }
 
+    @Override
+    public void deleteFiles(Collection<String> paths)
+            throws IOException
+    {
+        Map<Path, List<Path>> pathsGroupedByDirectory = paths.stream().collect(
+                groupingBy(path -> hadoopPath(path.replaceFirst("/[^/]*$", "")),
+                        mapping(HadoopPaths::hadoopPath, toList())));
+        for (Map.Entry<Path, List<Path>> directoryWithPaths : pathsGroupedByDirectory.entrySet()) {
+            FileSystem fileSystem = environment.getFileSystem(context, directoryWithPaths.getKey());
+            FileSystem rawFileSystem = getRawFileSystem(fileSystem);
```

Comment:
Inline. This would help understand that unwrapped file system is used in batched mode, but it's not used otherwise.